### PR TITLE
Update NEWS.md

### PR DESCRIPTION
**Pull request by MichaelChirico**
One final pass at tidying up the news: #6166

### CODE REVIEW

**Review by TysonStanley — APPROVED** [review]
Think this is looking nice. One thought; for the `dt[, (names(.SD)) := ...` did we want to also show it works with .SDcols? Or just include that in a vignette?

File: NEWS.md [hunk 1]
```diff
@@ -44,47 +44,64 @@
...
+
+    b. Prints the indicator `---` in every value column when truncation is needed and `row.names = FALSE` instead of adding a blank column where the `rownames` would have been just to include `---`, [#4083](https://github.com/Rdatatable/data.table/issues/4083). Thanks @MichaelChirico for the report and @joshhwuu for the fix.
+
+    c. Honors `na.print`, as seen in `print.default`, allowing for string replacement of `NA` values when printing. Thanks @HughParsonage for the report and @joshhwuu for the fix.
+
+    d. Gains new argument `show.indices` (with corresponding option `datatable.show.indices`) that allows the user to print a `data.table`'s indices as columns without having to modify the `data.table` itself. Thanks @MichaelChirico for the report and @joshhwuu for the PR.
+
+    e. Displays `integer64` columns well even if {bit64} had not yet been loaded, [#6224](https://github.com/Rdatatable/data.table/issues/6224). Thanks @renkun-ken for the report and @MichaelChirico for the fix.
```
Comment (Anirban166) [inline NEWS.md:77]: Suggestion: 'e. Displays `integer64` columns even if {bit64} has not yet been loaded ... '

File: NEWS.md [hunk 2]
```diff
@@ -185,13 +195,13 @@
...
+18. `?melt` has long documented that the returned `variable` column should contain integer column indices when `measure.vars` is a list, but when the list length is 1, `variable` is actually a character column name, which is inconsistent with the documentation, [#5209](https://github.com/Rdatatable/data.table/issues/5209). To increase consistency in the next release, we plan to change `variable` to integer, so users who were relying on this behavior should change `measure.vars=list("col_name")` (output `variable` is column name, will be column index/integer) to `measure.vars="col_name"` (`variable` is column name before and after the planned change). For now, relying on this undocumented behavior throws a new warning.
```
Comment (Anirban166) [inline NEWS.md:202]: > (output `variable` is column name, will be column index/integer)

(`variable` currently outputs a column name but will now become a column index/integer)

> (`variable` is column name before and after the planned change)

(`variable` will remain as a column name)

**Review by Anirban166 — COMMENTED** [review]
LGTM! A lot of great stuff.

**Review by Anirban166 — APPROVED** [review]

File: NEWS.md (hunk 1 above)
Comment (MichaelChirico) [reply NEWS.md:77]: How about 'Displays `integer64` columns correctly by loading {bit64} if needed'

File: NEWS.md (hunk 2 above)
Comment (MichaelChirico) [reply NEWS.md:202]: I agree this wording is not totally clear but I can't confirm what's right at the moment, @tdhock could you take another look?

Comment (tdhock) [reply NEWS.md:202]: how about this? I think the wording "before and after the planned change" is clear.
 (variable currently is a column name but will be a column index/integer after the planned change)
(variable is column name before and after the planned change)

**Review by tdhock — APPROVED** [review]

**Comment by MichaelChirico** [comment]
Thanks all!

**Pull request merged**